rest/src/main/groovy/se/kb/libris/digi/DigitalReproductionAPI.groovy

```diff
@@ -261,19 +272,24 @@ class ReproductionService {
         thing.hasRepresentation || !asList(getAtPath(thing, ['associatedMedia', '*', 'uri'], [])).isEmpty()
     }
     
-    void createHoldingFor(thingId, heldById) {
+    void createHoldingFor(thingId, Map item) {
+        def record = asMap(item.remove('meta'))
+        def heldById = getAtPath(item, ['heldBy', '@id'])
+        
+        def components = (asList(item.hasComponent) ?: [[:]]).collect {
+            it + [
+                    '@type' : 'Item',
+                    'heldBy': ['@id' : heldById],
+            ]
+        }
+        
         xl.create(
-                [:],
-                [
+                record,
+                item + [
                         '@type' : 'Item',
                         'itemOf' : ['@id' : thingId],
                         'heldBy': ['@id' : heldById],
-                        'hasComponent' : [
-                                [
-                                        '@type' : 'Item',
-                                        'heldBy': ['@id' : heldById],
-                                ]
-                        ]
+                        'hasComponent' : components
```

Comment:
Only add hasComponent if non-empty? (RDF-wise an empty set means no arc, but that also makes it superfluous here.)

Reply:
Had a feeling I did this a bit too "clever" =P
There is always at least one hasComponent. Since this is Libris standard practice.

`(asList(item.hasComponent) ?: [[:]])` <-- Line 279, a single empty component is added if missing

Reply:
Ah, I see it now! 😉 

(Alas, that standard practise is bad; but also terribly hard to get at (as the code has been built up around that).)